File: app/src/main/java/com/bapplications/maplemobile/opengl/utils/TimedBool.java
package com.bapplications.maplemobile.opengl.utils;

public class TimedBool {

    private long last;
    private long delay;
    private boolean value;
    
    public TimedBool()
    {
        value = false;
        delay = 0;
        last = 0;
    }

    public boolean isTrue()
    {
        return value;
    }

    public void set_for(long millis)
    {
        last = millis;
        delay = millis;
        value = true;
    }

    public void update(int deltatime)
    {
        if (value)
        {
            if (deltatime >= delay)
            {
                value = false;
                delay = 0;
            }
            else
            {
                delay -= deltatime;
            }
        }
    }

    public void set (boolean b)
    {
        value = b;
        delay = 0;
        last = 0;
    }

    public boolean equals (boolean b)
    {
        return value == b;
    }

    public boolean notEquals (boolean b)
    {
        return value != b;
    }

    public float alpha()
    {
        return 1.0f - (float)(delay) / last;
    }
}

File: app/src/main/java/com/bapplications/maplemobile/utils/TimedBool.kt
package com.bapplications.maplemobile.utils

class TimedBool {
    private var last: Long = 0
    private var delay: Long = 0
    var isTrue = false
        private set

    fun setFor(millis: Long) : TimedBool {
        last = millis
        delay = millis
        isTrue = true
        return this
    }

    fun update(deltatime: Int) {
        if (isTrue) {
            if (deltatime >= delay) {
                isTrue = false
                delay = 0
            } else {
                delay -= deltatime.toLong()
            }
        }
    }

    fun set(b: Boolean) {
        isTrue = b
        delay = 0
        last = 0
    }

    fun equals(b: Boolean): Boolean {
        return isTrue == b
    }

    fun alpha(): Float {
        return 1.0f - delay.toFloat() / last
    }
}